tls/s2n_handshake_io.c

```diff
@@ -1591,7 +1591,7 @@ static int s2n_handle_retry_state(struct s2n_connection *conn)
 
 bool s2n_handshake_is_complete(struct s2n_connection *conn)
 {
-    return conn && ACTIVE_STATE(conn).writer == 'B';
+    return conn && (ACTIVE_STATE(conn).writer == 'B' || conn->deserialized_conn);
```

Comment:
Might need a quick comment about why "deserialized_conn" implies that the handshake is complete